test.js

```diff
@@ -329,6 +329,25 @@ test('Custom health check', t => {
       })
     })
   })
+
+  t.test('should wait for the initial healthCheck call before initialising the server', t => {
+    t.plan(3)
+
+    let called = false
+    const fastify = Fastify()
+    fastify.register(underPressure, {
+      healthCheck: async () => {
+        t.false(called)
+        called = true
+      }
+    })
+
+    fastify.listen(0, (err) => {
+      t.error(err)
+      t.true(called)
+      fastify.close()
+    })
+  })
```

Comment:
would this test fail without the fix in the code? You might have to add a `sleep` in the healthCheck.

Reply:
I added a promisified wait so the sleep doesn't block the eventloop to allow other async components to initialise to ensure that it is just that check blocking starting